core/src/main/scala/com/yahoo/maha/core/query/druid/DruidQueryGenerator.scala

```diff
@@ -298,11 +297,22 @@ class DruidQueryGenerator(queryOptimizer: DruidQueryOptimizer
     return requestModel.startIndex < maximumMaxRows;
   }
 
+  def createDataSource(dataSource: String): DataSource = {
+    if(dataSource.split(",").length <= 1)
+      new TableDataSource(dataSource.split(",").head)
```

Comment:
is this the same return type as `fact.underlyingTableName.getOrElse(fact.name)`?
if not, will this new return type affect any existing behavior?

Reply:
This converts all input from String to DataSource, which is one of the valid use cases as input to the below function (it accepts String type or dataSource type)
Here's the actual code:
https://github.com/apache/druid/blob/master/processing/src/main/java/org/apache/druid/query/topn/TopNQueryBuilder.java#L158

https://github.com/apache/druid/blob/master/processing/src/main/java/org/apache/druid/query/topn/TopNQueryBuilder.java#L173

String input casts to TableDataSource, DataSource input allows use of any DataSource type.

Reply:
Sounds good - builder accepts both String and DataSource type is the answer I'm looking for.